File: DigitalIcebreakers.EndToEndTests/Helpers/Presenter.cs
using System;
using System.Threading.Tasks;
using PlaywrightSharp;
using PlaywrightSharp.Chromium;

namespace DigitalIcebreakers.EndToEndTests
{
    public class Presenter : AbstractBrowser
    {
        public readonly string _url;
        public string Url => _url;
        public Presenter(IBrowser browser, IPage page, string url) : base(browser, page)
        {
            _url = url;
        }

        private async Task StartGame(string gameId)
        {
            await _page.ClickAsync("text='New Activity'");
            await _page.ClickByTestId($"game-{gameId}");
        }

        public async Task LoadTriviaQuestions(string json = "questions.json")
        {
            await StartTrivia();
            await _page.ClickAsync("text='Questions'");
            var element = await _page.QuerySelectorAsync("[type=file]");
            await element.SetInputFilesAsync($"./tests/Trivia/{json}");
            await _page.ClickAsync("text='Trivia'");
        }

        public async Task LoadPollQuestions(string json = "questions.json")
        {
            await StartPoll();
            await _page.ClickAsync("text='Questions'");
            var element = await _page.QuerySelectorAsync("[type=file]");
            await element.SetInputFilesAsync($"./tests/Trivia/{json}");
            await _page.ClickAsync("text='Poll'");
        }

        public async Task StartBroadcast()
        {
            await StartGame("broadcast");
        }

        public async Task StartTrivia()
        {
            await StartGame("trivia");
        }

        public async Task StartPoll()
        {
            await StartGame("poll");
        }

        public async Task StartPong()
        {
            await StartGame("pong");
        }
    }
}

File: web/e2e/helpers/presenter.ts
import { Browser, Page } from "@playwright/test";
import { AbstractBrowser } from "./abstractBrowser";
import path from "path";
import { fileURLToPath } from "url";

export class Presenter extends AbstractBrowser {
  constructor(browser: Browser, page: Page, public readonly url: string) {
    super(browser, page);
  }

  private async startGame(gameId: string): Promise<void> {
    await this.page.getByRole("link", { name: "New Activity" }).click();
    await this.page.getByTestId(`game-${gameId}`).click();
  }

  async loadTriviaQuestions(json: string = "questions.json"): Promise<void> {
    await this.startTrivia();
    await this.page.getByRole("link", { name: "Questions" }).click();
    const fileInput = await this.page.locator("[type=file]");
    const __dirname = path.dirname(fileURLToPath(import.meta.url));
    const filePath = path.resolve(__dirname, "../test-data/trivia", json);
    await fileInput.setInputFiles(filePath);
    await this.page.getByRole("link", { name: "Trivia" }).click();
  }

  async loadPollQuestions(json: string = "questions.json"): Promise<void> {
    await this.startPoll();
    await this.page.getByRole("link", { name: "Questions" }).click();
    const fileInput = await this.page.locator("[type=file]");
    const __dirname = path.dirname(fileURLToPath(import.meta.url));
    const filePath = path.resolve(__dirname, "../test-data/trivia", json);
    await fileInput.setInputFiles(filePath);
    await this.page.getByRole("link", { name: "Poll" }).click();
  }

  async startBroadcast(): Promise<void> {
    await this.startGame("broadcast");
  }

  async startRetrospective(): Promise<void> {
    await this.startGame("retrospective");
  }

  async startFistOfFive(): Promise<void> {
    await this.startGame("fist-of-five");
  }

  async startTrivia(): Promise<void> {
    await this.startGame("trivia");
  }

  async startPoll(): Promise<void> {
    await this.startGame("poll");
  }

  async startPong(): Promise<void> {
    await this.startGame("pong");
  }

  async startYesNoMaybe(): Promise<void> {
    await this.startGame("yes-no-maybe");
  }

  async startNamePicker(): Promise<void> {
    await this.startGame("name-picker");
  }

  async startReaction(): Promise<void> {
    await this.startGame("reaction");
  }

  async startIdeaWall(): Promise<void> {
    await this.startGame("idea-wall");
  }
}
